### PR TITLE
Support reinstalling canisters (SNS)

### PR DESCRIPTION
**Pull request by anchpop**
# Description

Previously there was no way to send a UpgradeSnsControlledCanister proposal that reinstalls a canister with quill. This MR adds a `--mode` flag that can be used for this purpose. The flag is required. 

Fixes [NNS1-3063](https://dfinity.atlassian.net/browse/NNS1-3063)

# How Has This Been Tested?

Automated tests have been adjusted to test all 3 cases 

# Checklist

- [x] I have made corresponding changes to the documentation in docs/cli-reference.

[NNS1-3063]: https://dfinity.atlassian.net/browse/NNS1-3063?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/commands/sns/make_upgrade_canister_proposal.rs [hunk 1]
```diff
@@ -62,6 +63,11 @@ pub struct MakeUpgradeCanisterProposalOpts {
     /// Path to the binary file containing argument to post-upgrade method of the new canister WASM.
     #[arg(long, conflicts_with = "canister_upgrade_arg")]
     canister_upgrade_arg_path: Option<String>,
+
+    /// If true, the canister will be reinstalled instead of upgraded.
```
Comment (sesi200) [inline src/commands/sns/make_upgrade_canister_proposal.rs:67]: Outdated

**Review by sesi200 — COMMENTED** [review]
LGTM

Can you also add a changelog entry and update the PR description please?

File: src/commands/sns/make_upgrade_canister_proposal.rs (hunk 1 above)
Comment (anchpop) [reply src/commands/sns/make_upgrade_canister_proposal.rs]: Fixed!

**Review by sesi200 — DISMISSED** [review]

File: src/commands/sns/make_upgrade_canister_proposal.rs [hunk 2]
```diff
@@ -62,6 +63,11 @@ pub struct MakeUpgradeCanisterProposalOpts {
     /// Path to the binary file containing argument to post-upgrade method of the new canister WASM.
     #[arg(long, conflicts_with = "canister_upgrade_arg")]
     canister_upgrade_arg_path: Option<String>,
+
+    /// The install mode.
+    #[arg(long, short, default_value("install"),
```
Comment (sesi200) [inline src/commands/sns/make_upgrade_canister_proposal.rs:68]: This disagrees with `If the flag is not passed, the canister is upgraded instead (the old behavior).` from the PR description. Also, it changes the default behaviour. IMO we need a very good reason to change what it does by default

File: src/commands/sns/make_upgrade_canister_proposal.rs [hunk 3]
```diff
@@ -27,8 +28,8 @@ pub struct MakeUpgradeCanisterProposalOpts {
     proposer_neuron_id: ParsedSnsNeuron,
 
     /// Title of the proposal.
-    #[arg(long, default_value_t = String::from("Upgrade Canister"))]
-    title: String,
+    #[clap(long)]
```
Comment (adamspofford-dfinity) [inline src/commands/sns/make_upgrade_canister_proposal.rs:31]:
```suggestion
    #[arg(long)]
```

File: src/commands/sns/make_upgrade_canister_proposal.rs [hunk 4]
```diff
@@ -62,6 +63,10 @@ pub struct MakeUpgradeCanisterProposalOpts {
     /// Path to the binary file containing argument to post-upgrade method of the new canister WASM.
     #[arg(long, conflicts_with = "canister_upgrade_arg")]
     canister_upgrade_arg_path: Option<String>,
+
+    /// The install mode.
+    #[arg(long, short, value_parser = ["install", "reinstall", "upgrade"])]
```
Comment (adamspofford-dfinity) [inline src/commands/sns/make_upgrade_canister_proposal.rs:68]: Quill tries to be very conservative about short flags. I don't think we have any besides -y and neuron-manage's -a.

```suggestion
    #[arg(long, value_parser = ["install", "reinstall", "upgrade"])]
```

**Review by sesi200 — APPROVED** [review]

**Pull request merged**